packages/taiko-client/pkg/rpc/methods.go

```diff
@@ -374,23 +374,50 @@ func (c *Client) GetPoolContent(
 	ctxWithTimeout, cancel := CtxWithTimeoutOrDefault(ctx, defaultTimeout)
 	defer cancel()
 
-	l1Head, err := c.L1.HeaderByNumber(ctx, nil)
+	// Get the latest L2 block header at first.
+	l2Head, err := c.L2.HeaderByNumber(ctx, nil)
 	if err != nil {
 		return nil, err
 	}
 
-	l2Head, err := c.L2.HeaderByNumber(ctx, nil)
+	l1Origin, err := c.L2.L1OriginByID(ctx, l2Head.Number)
+	if err != nil && err.Error() != ethereum.NotFound.Error() {
```

Comment:
Shouldn't `err` be returned directly? Otherwise, panic is likely to occur below.

Reply:
`l1Origin != nil` is checked below, don't think it will panic

Reply:
And if there is no soft block, there will be `ethereum.NotFound` error, which is by design